api/v1alpha1/oadp_types.go

```diff
@@ -396,20 +396,23 @@ func (dpa *DataProtectionApplication) AutoCorrect() {
 		// duplicate feature flag checks are done in ReconcileVeleroDeployment
 		dpa.Spec.Configuration.Velero.FeatureFlags = append(dpa.Spec.Configuration.Velero.FeatureFlags, velero.APIGroupVersionsFeatureFlag)
 	}
-	if dpa.Spec.Configuration.Velero.Args != nil {
-		// if args is not nil, we take care of some fields that will be overridden from dpa if not specified in args
-		// Enable user to specify --fs-backup-timeout duration (OADP default 1h0m0s)
-		resticTimeout := "1h"
-		if dpa.Spec.Configuration != nil && dpa.Spec.Configuration.Restic != nil && len(dpa.Spec.Configuration.Restic.Timeout) > 0 {
-			resticTimeout = dpa.Spec.Configuration.Restic.Timeout
-		}
-		if pvOperationTimeout, err := time.ParseDuration(resticTimeout); err == nil && dpa.Spec.Configuration.Velero.Args.PodVolumeOperationTimeout == nil {
-			dpa.Spec.Configuration.Velero.Args.PodVolumeOperationTimeout = &pvOperationTimeout
-		}
+
+	if dpa.Spec.Configuration.Velero.Args == nil {
+		dpa.Spec.Configuration.Velero.Args = &server.Args{}
```

Comment:
```suggestion
```
Can we do this without this line?

This would cause https://github.com/openshift/oadp-operator/pull/1139/files#diff-992e40b76fc7cf12af63ec6c2cdcc70321669228469781c0d2bc527dc4c28627R436 path to run everytime in velero.go
```go
	// if server args is set, override the default server args
	if dpa.Spec.Configuration.Velero.Args != nil {
```

We only wanted to help fill dpa.Spec.Configuration.Velero.Args if user themselves have made velero args not nil, otherwise use default velero functions to fill velero container args.

As is, we would be overriding args defined earlier in velero.go such as
```
veleroContainer.Args = append(veleroContainer.Args, fmt.Sprintf("--fs-backup-timeout=%s", fsBackupTimeout))
```

Reply:
I think not, as Scott pointed https://github.com/openshift/oadp-operator/pull/1139#discussion_r1305782591 without this, we can hit nil pointer

But yes, this is causing that to run everytime, I was even going to ask if we can change from
```go
        // Enable user to specify --fs-backup-timeout (defaults to 1h)
	fsBackupTimeout := "1h"
	if dpa.Spec.Configuration.Restic != nil && len(dpa.Spec.Configuration.Restic.Timeout) > 0 {
		fsBackupTimeout = dpa.Spec.Configuration.Restic.Timeout
	}
	// Append restic timeout option manually. Not configurable via install package, missing from podTemplateConfig struct. See: https://github.com/vmware-tanzu/velero/blob/8d57215ded1aa91cdea2cf091d60e072ce3f340f/pkg/install/deployment.go#L34-L45
	veleroContainer.Args = append(veleroContainer.Args, fmt.Sprintf("--fs-backup-timeout=%s", fsBackupTimeout))

	setContainerDefaults(veleroContainer)
	// if server args is set, override the default server args
	if dpa.Spec.Configuration.Velero.Args != nil {
		var err error
		veleroContainer.Args, err = dpa.Spec.Configuration.Velero.Args.StringArr(dpa.Spec.Configuration.Velero.FeatureFlags)
		if err != nil {
			return err
		}
	}
	return nil
```

to
```go

	setContainerDefaults(veleroContainer)
	var err error
	veleroContainer.Args, err = dpa.Spec.Configuration.Velero.Args.StringArr(dpa.Spec.Configuration.Velero.FeatureFlags)
	if err != nil {
		return err
	}
	return nil
```

Reply:
Going to fix this in next commit